pkg/transport/process/extensions/uds_executable.go

```diff
@@ -0,0 +1,118 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+package extensions
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/exec"
+	"syscall"
+	"time"
+
+	"github.com/gardener/component-cli/pkg/transport/process"
+	"github.com/gardener/component-cli/pkg/utils"
+)
+
+// ServerAddressEnv is the environment variable key which is used to store the
+// address under which a resource processor server should start.
+const ServerAddressEnv = "SERVER_ADDRESS"
+
+type udsExecutable struct {
+	bin  string
+	args []string
+	env  []string
+	addr string
+}
+
+// NewUDSExecutable runs a resource processor extension executable in the background.
```

Comment:
To clarify: Is the udsExecutable already running or will this happen in Process? Same for stdExecutable